src/identity/quartzOrganizations/actions/thunks/index.ts

```diff
@@ -13,73 +16,95 @@ import {
 import {PublishNotificationAction} from 'src/shared/actions/notifications'
 
 // Types
-import {GetState, RemoteDataState} from 'src/types'
+import {AppThunk, GetState, RemoteDataState} from 'src/types'
 import {OrganizationSummaries} from 'src/client/unityRoutes'
-
 type Actions = QuartzOrganizationActions | PublishNotificationAction
 type DefaultOrg = OrganizationSummaries[number]
 
-class OrgNotFoundError extends Error {
+interface UpdateOrgParams {
+  accountId: number
+  newDefaultOrg: DefaultOrg
+}
+
+// Utils
+import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
+
+export enum ThunkErrorNames {
+  DefaultOrgStateError = 'DefaultOrgStateError',
+  DefaultOrgNetworkError = 'DefaultOrgNetworkError',
+}
+
+export class DefaultOrgStateError extends Error {
   constructor(message) {
     super(message)
-    this.name = 'DefaultOrgNotFoundError'
+    this.name = ThunkErrorNames.DefaultOrgStateError
   }
 }
 
-// Error Reporting
-import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
+export class DefaultOrgNetworkError extends Error {
+  constructor(message) {
+    super(message)
+    this.name = ThunkErrorNames.DefaultOrgNetworkError
+  }
+}
 
-export const getQuartzOrganizationsThunk = () => async (
+// Thunks
+export const getQuartzOrganizationsThunk = (accountId: number) => async (
   dispatch: Dispatch<Actions>,
   getState: GetState
 ) => {
   try {
     dispatch(setQuartzOrganizationsStatus(RemoteDataState.Loading))
-    const quartzOrganizations = await fetchQuartzOrgs()
+
+    const quartzOrganizations = await fetchOrgsByAccountID(accountId)
 
     dispatch(setQuartzOrganizations(quartzOrganizations))
   } catch (err) {
-    dispatch(setQuartzOrganizationsStatus(RemoteDataState.Error))
-
     reportErrorThroughHoneyBadger(err, {
       name: 'Failed to fetch /quartz/orgs/',
       context: {state: getState()},
     })
   }
 }
 
-export const updateDefaultOrgThunk = (newDefaultOrg: DefaultOrg) => async (
+export const updateDefaultOrgThunk = ({
+  accountId,
+  newDefaultOrg,
+}: UpdateOrgParams): AppThunk<Promise<void>> => async (
   dispatch: Dispatch<Actions>,
   getState: GetState
-) => {
+): Promise<void> => {
   try {
     dispatch(setQuartzOrganizationsStatus(RemoteDataState.Loading))
 
-    await updateDefaultQuartzOrg(newDefaultOrg.id)
+    await updateDefaultOrgByAccountID({
+      accountNum: accountId,
+      orgId: newDefaultOrg.id,
+    })
 
     dispatch(setQuartzDefaultOrg(newDefaultOrg.id))
 
     const state = getState()
     const orgStatus = state.identity.currentIdentity.status
 
     if (orgStatus === RemoteDataState.Error) {
-      const defaultOrgErrMsg =
```

Comment:
Removed this, because by throwing an error in the case of a state management problem, we can make a single call to `reportErrorThroughHoneyBadger` in the catch block whether the error is state-based or network-based, and `err.name` will report the correct error.